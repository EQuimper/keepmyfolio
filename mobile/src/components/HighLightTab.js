// @flow

import React from 'react';
import { StyleSheet, View } from 'react-native';

import { colors } from '../utils/constants';

const styles = StyleSheet.create({
  root: {
    alignItems: 'center',
    alignSelf: 'stretch',
    borderRadius: 1,
    flex: 1,
    justifyContent: 'center',
  },
});

type Props = {
  children: React.Element<*>,
  focused: boolean,
};

function HighLightTab({ children, focused }: Props) {
  let footerHeight;

  if (focused) {
    footerHeight = 1;
  } else {
    footerHeight = 0;
  }

  return (
    <View
      footerHeight={footerHeight}
      style={[
        styles.root,
        { borderBottomColor: colors.primary, borderBottomWidth: footerHeight },
      ]}
    >
      {children}
    </View>
  );
}

export default HighLightTab;
