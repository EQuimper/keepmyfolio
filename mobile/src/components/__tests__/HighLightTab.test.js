// @noflow

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import HighLightTab from '../HighLightTab';

describe('HighLightTab Components', () => {
  it('should renders correctly', () => {
    const component = renderer.create(<HighLightTab focused />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should have a borderBottomWidth of 1 if focused', () => {
    const component = renderer.create(<HighLightTab focused />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should have a borderBottomWidth of 0 if focused is false', () => {
    const component = renderer.create(<HighLightTab focused={false} />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
