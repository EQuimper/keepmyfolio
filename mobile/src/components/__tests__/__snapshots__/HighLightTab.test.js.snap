// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighLightTab Components should have a borderBottomWidth of 0 if focused is false 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 1,
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "borderBottomColor": "#00C0FF",
        "borderBottomWidth": 0,
      },
    ]
  }
/>
`;

exports[`HighLightTab Components should have a borderBottomWidth of 1 if focused 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 1,
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "borderBottomColor": "#00C0FF",
        "borderBottomWidth": 1,
      },
    ]
  }
/>
`;

exports[`HighLightTab Components should renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 1,
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "borderBottomColor": "#00C0FF",
        "borderBottomWidth": 1,
      },
    ]
  }
/>
`;
